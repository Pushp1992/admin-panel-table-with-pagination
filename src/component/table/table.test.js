import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Table from './table';
import '@testing-library/jest-dom';
import { data } from './mock-data';

afterEach(() => {
    cleanup()
});

it('match snapshot with DOM tree', () => {
    const component = renderer.create(<Table/>);
    const DOMTree = component.toJSON();
    expect(DOMTree).toMatchSnapshot();
});

it('renders Table Component', () => {
    render(<Table items={data} />);
    const tableElement = screen.getAllByTestId('table');
    expect(tableElement[0]).toBeInTheDocument();
});

it('renders single table', () => {
    render(<Table items={data} />);
    const tableElement = screen.getAllByTestId('table');
    expect(tableElement).toHaveLength(1);
});