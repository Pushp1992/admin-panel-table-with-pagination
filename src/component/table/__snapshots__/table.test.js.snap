// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot with DOM tree 1`] = `
<div
  className="TableWrapper-cTmqaQ liBAoc"
>
  <div
    className="DeleteAllButtonWrpper-SBInx eIewal"
  />
  <table
    data-testid="table"
  >
    <tr
      data-testid="table-head"
    >
      <th>
        <input
          className="select-all"
          data-testid="checkbox"
          onChange={[Function]}
          type="checkbox"
        />
        <label />
      </th>
      <th>
        #Id
      </th>
      <th>
        Name
      </th>
      <th>
        Email
      </th>
      <th>
        Role
      </th>
    </tr>
  </table>
</div>
`;
